language-support/codegen-common/src/main/scala/com/digitalasset/daml/lf/codegen/conf/CodegenConfigReader.scala

```diff
@@ -157,13 +157,15 @@ object CodegenConfigReader {
 
   implicit val decodePackageReference: KeyDecoder[PackageReference] =
     new KeyDecoder[PackageReference] {
+      // This regex matches IdString.PackageVersion.
+      private val refRegex = """^(.*)-((0|[1-9][0-9]*)(\.(0|[1-9][0-9]*))*)$""".r
```

Comment:
This worries me a bit. It also statically assumes a specific format for the version number, which we probably don't want us to tie ourselves to. Given the code that you edited below (unless the comment I made applies), it appears that we can assume that the last component separated by a `-` is the version. Can we do that instead?

Reply:
This is specified in daml-lf so we _need_ to tie ourselves to that.

Reply:
I might have misunderstood your point. I switched to splitting on the last dash which doesn’t tie us any less to the format (PackageVersion.fromString will explode) but it removes the duplication which seem snice I guess